### PR TITLE
feat: bring postgres

### PR DESCRIPTION
**Pull request by Chriscbr**
Introduces a postgres `Database` resource. On cloud targets it uses [Neon](https://neon.tech/). Currently it only works on AWS because I want to inject a secret during deployment, and the current `cloud.Secret` resource doesn't support that, but it should be possible to make the cloud implementation cloud agnostic (so it works on GCP and Azure as well).

The simulator target should be able to use an ordinary postgres container, but I haven't implemented it yet as I think I need to use the `containers` library, which will introduces peer dependencies in this repo and... yeah.

Closes https://github.com/winglang/wing/issues/4914

### CODE REVIEW

File: postgres/lib.w [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+bring "@cdktf/provider-aws" as tfaws;
+
+struct Credentials {
+  host: str;
+  user: str;
+  password: str;
+  dbname: str;
+}
+
+pub struct DatabaseNeonProps {
```
Comment (eladb) [inline postgres/lib.w:16]:
```suggestion
pub struct DatabaseProps {
```

File: postgres/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+## `sim`
+
+When executed in the Wing Simulator, postgres is run within a local Docker container.
+
+## `tf-aws`
+
+On the `tf-aws` target, the postgres database is managed using [Neon](https://neon.tech/), a serverless postgres offering.
+
+Neon has a [free tier](https://neon.tech/docs/introduction/free-tier) that can be used for personal projects and prototyping.
```
[the remark this answers is not in the document]
Comment (Chriscbr) [reply postgres/README.md:42]: Good question - yeah, I didn't research the options too closely. I was mostly just curious to try out some third-party cloud resources. One upside of using Neon could be that it guarantees if they switch their primary platform to Azure or GCP, the database parts of their app will work the same (maybe they won't even have to perform a migration!) But yeah, I love the idea of allowing users to choose their own dependency injections as needed

**Comment by Chriscbr** [comment]
Uh oh, looks like pull requests are publishing to npm. Investigating...

**Comment by eladb** [comment]
> Uh oh, looks like pull requests are publishing to npm. Investigating...


oops

**Pull request merged**